pkg/api/handlers/compat/containers.go

```diff
@@ -87,7 +87,7 @@ func ListContainers(w http.ResponseWriter, r *http.Request) {
 		utils.InternalServerError(w, err)
 		return
 	}
-	if _, found := r.URL.Query()["limit"]; found {
+	if _, found := r.URL.Query()["limit"]; found && query.Limit != -1 {
```

Comment:
Wouldn't `>= 0` be safer?

Reply:
Yes, then if user sends any negative number it will work.